src/main/scala/devices/debug/Debug.scala

```diff
@@ -281,6 +288,7 @@ class TLDebugModuleOuter(device: Device)(implicit p: Parameters) extends LazyMod
         DMCONTROLNxt.haltreq      := DMCONTROLWrData.haltreq
         DMCONTROLNxt.resumereq    := DMCONTROLWrData.resumereq
         DMCONTROLNxt.ackhavereset := DMCONTROLWrData.ackhavereset
+        DMCONTROLNxt.hasel        := supportHartArray.B & DMCONTROLWrData.hasel
```

Comment:
can we keep the logic in the chisel domain, vs generating logic for things we know at elaboration time? I'd write this as:
```
DMCONTROLNxt.hasel := if supportHartArray DMCONTROLWrData.hasel else false.B
```

Reply:
I couldn't (and still can't) get something like this (where scala is embedded within a line of chisel) to compile.  The logic is removed eventually since supportHartArray is a constant, but I agree not generating any logic in the first place would be preferable.  Any suggestions?

Reply:
```
[error] /scratch/erniee/federation/rocket-chip/src/main/scala/devices/debug/Debug.scala:291:38: illegal start of simple expression
[error]         DMCONTROLNxt.hasel        := if supportHartArray DMCONTROLWrData.hasel else false.B
[error]                                      ^
[error] one error found

```

Reply:
try throwing parentheses around it... `DMCONTROLNxt.hasel := (if ...)`

Reply:
Oh, also maybe `if (supportsHartArray) ... `

Reply:
Parens around the entire if statement worked...makes sense!
`DMCONTROLNxt.hasel        := (if (supportHartArray) DMCONTROLWrData.hasel else false.B)`